pyrobosim/pyrobosim/manipulation/grasping.py

```diff
@@ -0,0 +1,182 @@
+""" Grasping utilities. """
+
+from ..utils.pose import Pose
+
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+from transforms3d.quaternions import rotate_vector, qinverse
+
+
+class Grasp:
+    """
+    Representation of an object grasp.
+    """
+    def __init__(self, origin, direction, properties=None):
+        self.origin = origin
+        self.direction = direction
+        self.properties = properties
+
+    def add_origin(self, vec):
```

Comment:
This seems to describe what's happening better:

```suggestion
    def translate_origin(self, vec):
```